SpiFlash.class.nut

```diff
@@ -70,106 +63,101 @@ class SPIFlash {
         _spi_wr = _spi.writeread.bindenv(spi);
         _millis = hardware.millis.bindenv(hardware);
 
-        // Make sure we have SPIFLASH_PREVERIFY and SPIFLASH_POSTVERIFY defined and accurate
-        try {
-            _SPIFLASH_PREVERIFY = SPIFLASH_PREVERIFY;
-            _SPIFLASH_POSTVERIFY = SPIFLASH_POSTVERIFY;
-        } catch (e) { }
+        // We can safely configure the GPIO lines
+        _cs_l.configure(DIGITAL_OUT, 1);
     }
 
-    // -------------------------------------------------------------------------
     // spiflash.configure() - [optional] configures the SPI lines
     function configure(speed = 15000) {
         return _spi.configure(CLOCK_IDLE_LOW | MSB_FIRST, speed);
     }
 
-    // -------------------------------------------------------------------------
-    // spiflash.size() â€“ Returns the total number of bytes in the SPI flash that are available to Squirrel.
+    // spiflash.size() - Returns the total number of bytes in the SPI flash that are available to Squirrel.
     function size() {
-        return _blocks * BLOCK_SIZE;
+        // Throw error if disabled
+        if (!_enabled) throw SPI_NOT_ENABLED;
+
+        return _blocks * ELECTRICIMP_SPIFLASH_BLOCK_SIZE;
     }
 
-    // -------------------------------------------------------------------------
-    // spiflash.disable() â€“ Disables the SPI flash for reading and writing.
+    // spiflash.disable() - Disables the SPI flash for reading and writing.
     function disable() {
-        if (!_enabled) throw SPI_NOT_ENABLED;
+        // if we're already disabled, return
+        if(!_enabled) return;
+
         _enabled = false;
 
         _cs_l_w(0);
-        _spi_w(DP.tochar());
+        _spi_w(ELECTRICIMP_SPIFLASH_DP);
         _cs_l_w(1);
     }
 
-    // -------------------------------------------------------------------------
-    // spiflash.enable() â€“ Enables the SPI flash for reading and writing.
+    // spiflash.enable() - Enables the SPI flash for reading and writing.
     function enable() {
+        // If we're already enabled, return
+        if (_enabled) return;
+
         _enabled = true;
 
         _cs_l_w(0);
-        _spi_w(RDP.tochar());
+        _spi_w(ELECTRICIMP_SPIFLASH_RDP);
         _cs_l_w(1);
     }
```

Comment:
I think you have to give the SPI some time to initialize after you enable before you can issue a command. Might want to throw a little blocking sleep in here to enforce that.